src/MailChimp/Module/ModuleSubscribe.php

```diff
@@ -114,6 +107,47 @@ protected function compile()
         $this->Template->form = $form;
     }
 
+
+    /**
+     * Locates the position of the email field within the array of fields and inserts it
+     * @param $fields
+     * @return mixed
+     */
+    protected function insertEmailField($fields) {
+        $email = (object) [
+            'id' => 0,
+            'tag' => 'EMAIL',
+            'name' => $GLOBALS['TL_LANG']['tl_module']['mailchimp']['labelEmail'],
+            'type' => 'text',
+            'display_order' => -1,
+            'required' => true,
+            'public' => true,
+        ];
+
+        // if first field is displayed second, insert email field before
+        if ($fields[0]->display_order === 2) {
+            $email->display_order = 1;
+        } else {
+            // check if display order is consecutive
+            $index = $fields[0]->display_order;
+            foreach ($fields as $field) {
+                // if one display slot is missing, the email field goes here
+                if ($field->display_order === ($index + 1)) {
+                    $email->display_order = $index;
+                    break;
+                }
+                $index++;
+            }
+            // otherwise, append email field
+            if ($email->display_order === -1) {
```

Comment:
Just for CS reasons: please use yoda conditions `if (-1 === $email->displayOrder) {`